lib/Doctrine/CouchDB/HTTP/MultipartParserAndSender.php

```diff
@@ -0,0 +1,413 @@
+            $targetOptions['host'],
+            $targetOptions['port'],
+            $targetOptions['username'],
+            $targetOptions['password'],
+            $targetOptions['ip'],
+            $targetOptions['ssl']
+        );
+    }
+
+    /**
+     * Perform request to the source, parse the multipart response,
+     * stream the documents with attachments to the target and return
+     * the responses along with docs that did not have any attachments.
+     *
+     * @param string $sourceMethod
+     * @param string $sourcePath
+     * @param string $targetPath
+     * @param string $sourceData
+     * @param array $sourceHeaders
+     * @return array|ErrorResponse|string
+     * @throws HTTPException
+     * @throws \Exception
+     */
+    public function request(
+        $sourceMethod,
+        $sourcePath,
+        $targetPath,
+        $sourceData = null,
+        array $sourceHeaders = array()
+    ) {
+        $this->sourceConnection = $this->sourceClient->getConnection(
+            $sourceMethod,
+            $sourcePath,
+            $sourceData,
+            $sourceHeaders
+        );
+        $sourceResponseHeaders = $this->sourceClient->getStreamHeaders();
+        $body = '';
+
+        if (empty($sourceResponseHeaders['status'])) {
+            throw HTTPException::readFailure(
+                $this->sourceOptions['ip'],
+                $this->sourceOptions['port'],
+                'Received an empty response or not status code',
+                0
+            );
+
+        } elseif ($sourceResponseHeaders['status'] >= 400) {
+            while (!feof($this->sourceConnection)) {
+                $body .= fgets($this->sourceConnection);
+            }
+            return new ErrorResponse($sourceResponseHeaders['status'], $sourceResponseHeaders, $body);
+
+        } else {
+            if ($sourceResponseHeaders['status'] == 200) {
+
+                // Body is an array containing:
+                //  1) Array of json string documents that don't have attachments. These should be posted using the bulk
+                //     api.
+                //  2) Responses of posting docs with attachments.
+                $body = $this->parseAndSend($targetPath);
+            }
+            return $body;
+        }
+    }
+
+
+    /**
+     * Read and return next line from the connection pointer.
+     * $maxLength parameter can be used to set the maximum length
+     * to be read.
+     *
+     * @param int $maxLength
+     * @return string
+     */
+    protected function getNextLineFromSourceConnection($maxLength = null)
+    {
+        if ($maxLength !== null) {
+            return fgets($this->sourceConnection, $maxLength);
+        } else {
+            return fgets($this->sourceConnection);
+        }
+    }
+
+    /**
+     * Parses multipart data. Returns an array having:
+     * 1) Array of json docs(string) that don't have attachments. These
+     * should be posted using the bulk api. and
```

Comment:
Let's indent the second line for the first point.
Spell bulk api as "Bulk API"
Remove `. and` on the second line. The `1)` and `2)` points are enough for it to read well.